refactor: fix mypy issues in checkers/f*.py

Fix mypy errors for the following:

```
cve_bin_tool/checkers/file.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/file.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/fastd.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/fastd.py:16: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
```

Helpful resources:

- [Python typing documentation](https://docs.python.org/3/library/typing.html)
- [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
- If you can't find the answer in the docs, googling the error messages above will often help!
- [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md)
- If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:

- cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR. A good potential title for this one is in the title of this issue.
- You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/checkers/fastd.py
+++ b/cve_bin_tool/checkers/fastd.py
@@ -8,11 +8,13 @@ CVE checker for fastd
 https://www.cvedetails.com/product/90086/Fastd-Project-Fastd.html?vendor_id=23901
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class FastdChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = ["fastd v([0-9]+)"]
     VENDOR_PRODUCT = [("fastd_project", "fastd")]

--- a/cve_bin_tool/checkers/file.py
+++ b/cve_bin_tool/checkers/file.py
@@ -10,12 +10,14 @@ https://www.cvedetails.com/product/17327/Christos-Zoulas-File.html?vendor_id=199
 https://www.cvedetails.com/product/30576/File-Project-File.html?vendor_id=15034
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class FileChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [
         r"([0-9]+\.[0-9]+)\r?\n%s-%s\r?\nmagic file",
         r"([0-9]+\.[0-9]+)\r?\nmagic file",

